server-spi-private/src/main/java/org/keycloak/models/utils/DefaultKeyProviders.java

```diff
@@ -131,6 +132,25 @@ public static void createProviders(RealmModel realm, String privateKeyPem, Strin
             realm.addComponentModel(rsa);
         }
 
+        if (!hasProvider(realm, "rsa-enc")) {
+            ComponentModel rsaEnc = new ComponentModel();
+            rsaEnc.setName("rsa-enc");
+            rsaEnc.setParentId(realm.getId());
+            rsaEnc.setProviderId("rsa-enc");
+            rsaEnc.setProviderType(KeyProvider.class.getName());
+
+            MultivaluedHashMap<String, String> configEnc = new MultivaluedHashMap<>();
+            configEnc.putSingle("keyUse", KeyUse.ENC.getSpecName());
+            configEnc.putSingle("priority", "100");
+            configEnc.putSingle("privateKey", privateKeyPem);
+            if (certificatePem != null) {
+                configEnc.putSingle("certificate", certificatePem);
+            }
+            rsaEnc.setConfig(configEnc);
+
+            realm.addComponentModel(rsaEnc);
+        }
+
```

Comment:
I'm thinking we shouldn't do this, as it puts the server into an non-recommended setup where the same keys are used for signing and encryption.

In the backwards compatibility mode we just use the signature keys as previously, and we should just document how folks can create the proper encryption keys required using the component model in the realm representation, or through the UI.

Reply:
This code is used only with the deprecated [RealmRepresentation fields](https://github.com/keycloak/keycloak/blob/main/core/src/main/java/org/keycloak/representations/idm/RealmRepresentation.java#L101-L106). Removing this would mean we need to migrate all testsuite representations to use the new way of defining keys (using ComponentModel) or somehow hack the importing to double the keys before. 

I would probably prefer to leave this as is. Maybe we can mark this code as deprecated to make sure we remove it as part of the future issue I plan to create for removing the fields I mentioned above.